crypto/aes/aes_ige.c

```diff
@@ -38,15 +39,31 @@ void AES_ige_encrypt(const unsigned char *in, unsigned char *out,
                      size_t length, const AES_KEY *key,
                      unsigned char *ivec, const int enc)
 {
+    /* Ignore the return value since we are void */
+    int ret =AES_ige_encrypt_ex(in, out, length, key, ivec, enc);
+
+    assert(ret);
+}
+
+int AES_ige_encrypt_ex(const unsigned char *in, unsigned char *out,
+                       size_t length, const AES_KEY *key,
+                       unsigned char *ivec, const int enc)
+{
     size_t n;
     size_t len = length;
 
+    /* Nothing to do */
     if (length == 0)
-        return;
+        return 1;
 
-    OPENSSL_assert(in && out && key && ivec);
-    OPENSSL_assert((AES_ENCRYPT == enc) || (AES_DECRYPT == enc));
-    OPENSSL_assert((length % AES_BLOCK_SIZE) == 0);
+    if (in == NULL || out == NULL || key == NULL || ivec == NULL)
+        return 0;
```

Comment:
It can't be matter of coding style, because it's impossible to formulate good rule to go by. However, suggestion that it makes sense to check for NULL only when it's a recognized option does cover majority of cases. And maybe it should be mentioned in coding style as "first-one-to-consider". *Along with assert guidelines!* And as for NULL checks I see no shame in crashing when caller is responsible for validating input. And why just NULL, and not say `(void *)1`, `(void *)2`, etc. Why not check for alignment of key schedule?